### ISSUE TITLE
Remove [cross-fetch] dependency.

### ISSUE THREAD

**Issue by sosoba** [i18next/i18next-http-backend#104]
## 🚀 Feature Proposal

Remove `cross-fetch` dependency.

## Motivation

Nowadays Node 18 LTS has global `fetch` enabled by default.
Deno has also always had global `fetch`.

The `cross-fetch` has a hard coded `node-fetch` subdepndency which are redundant luggage. If someone need to support old Node versions they can add ponyfil directly in our legacy project:

```sh
npm i node-fetch
```
```js
if (!global.fetch) {
  global.fetch = require('node-fetch');
}
```

**Comment by adrai** [comment]
cross-fetch (node-fetch) is not used if global fetch is available...
so there should be no problem with that?

**Comment by sosoba** [comment]
The problem is the hard dependence on `node-fetch`. Npm always install thist package even hardly anyone needs it anymore.
This dependency should be optional.

**Comment by adrai** [comment]
And what's the problem if it's installed? It should not be bundled, right?

**Comment by adrai** [comment]
tldr;

- "hardly anyone needs it" => https://npmtrends.com/node-fetch
- fetch still experimental in v18 => https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch
- not everyone is on v18 right now
- still like to support older node versions without ponyfill
- is not bundled, only installed
- there is no such thing like conditional npm install i.e. install only if node < 18
- if there's no real issue with that, I would like to keep it as is... it may be in future this will change

**Comment by jonkoops** [comment]
@adrai now that the Fetch API has been marked as stable [in Node.js 21](https://nodejs.org/dist/latest-v21.x/docs/api/globals.html#fetch), and this API is available unflagged in all actively supported LTS versions (v18, v20), would you reconsider removing this dependency?

**Comment by adrai** [comment]
I still don't see any disadvantage of keeping it. Only because node-fetch or cross-fetch is installed in your node_modules, does not mean it gets loaded by the runtime or the browser.

btw: v2.3.0 uses the new cross-fetch update: https://github.com/lquixada/cross-fetch/releases/tag/v4.0.0 that officially included node v18 and v20 support

**Comment by jonkoops** [comment]
> I still don't see any disadvantage of keeping it. Only because node-fetch or cross-fetch is installed in your node_modules, does not mean it gets loaded by the runtime or the browser.

True, this would only remove the dependency from the package so that less code needs to be downloaded. I think that is still a valid thing to do, but perhaps should be part of a major version (if any will be made at some point). 

> btw: v2.3.0 uses the new cross-fetch update: https://github.com/lquixada/cross-fetch/releases/tag/v4.0.0 that officially included node v18 and v20 support

Interestingly it also drops support for Node.js 10 and 12 in that version. Are the supported Node.js versions for `18next-http-backend` documented somewhere (I can see 16, 18 and 20 [in the CI](https://github.com/i18next/i18next-http-backend/blob/226a70490bcad55b964f38ead251466b0922458a/.github/workflows/node.yml#L14))?

**Comment by jonkoops** [comment]
Just ran into this strange issue with `cross-fetch` and Vite due to it being a dependency of this package. It looks like with Vite 5 somehow both `cross-fetch` and `node-fetch` are included in production builds, which results in the following error:

> Uncaught ReferenceError: `global` is not defined

It appears that `cross-fetch` depends on an older version of `node-fetch`, which attempts to [access the `global` variable](https://github.com/node-fetch/node-fetch/blob/9b9d45881e5ca68757077726b3c0ecf8fdca1f29/src/index.js#L400) (that doesn't exist in browsers).

**Comment by adrai** [comment]
Can you please provide a reproducible example?

**Comment by jonkoops** [comment]
Sure, yeah working on that now.

**Comment by jonkoops** [comment]
Managed to reproduce this locally. Looks like this is due to some overrides we're setting in out Vite config, so I don't believe this is cause for concern.

**Comment by flipvh** [comment]
Hi @sosoba Sorry to bother you, but I came here looking to reduce my js bundle size a bit as I use this nice package in the browser. (Or should I not use this for frontend?) I noticed cross-fetch is quite a large dependency. Have you found a way to reduce bundle size yourself with this package or in another way? Thanks!!

**Comment by adrai** [comment]
ceoss-fetch is not loaded on runtime if native fetch implementation is available

**Comment by jacobbogers** [comment]
Hi, Can you expose the "fetch" as pluggable, this would help greatly at testing especially as we dont use mocks (it.concurrent will be an issue since jest mocks are global to test file) hence the need to make it pluggable

Example: 
```js
config: { 
    getFetch: () => globalThis.fetch
}
```

for mocking (integration  tests)

```js
config: { 
    getFetch: () => fetchMocked
}
```

**Comment by peter-at-work** [comment]
Since we only use `i18next-http-backend` in the frontend, and knowing that the outdated version of `node-fetch` will not be used (and `cross-fetch` almost feels abandoned), we put the following override in our _package.json_.  It also removes the _punycode_ warning that started appearing during the bundling process when the dev environment runs nodejs v21 or later.
```
  "overrides": {
    ...
    "i18next-http-backend": {
      "cross-fetch": {
        "node-fetch@^2": "^3"
      }
    },
    ...
  }
```

**Comment by adrai** [comment]
fyi: will remove the cross-fetch dep in the next major version

**Comment by jonkoops** [comment]
@adrai planning to go all in on a lean native fetch implementation?

**Comment by adrai** [comment]
fetch ponyfill will be completely removed

**Comment by adrai** [comment]
Just released `i18next-http-backend@4` — `cross-fetch` is gone. Native `fetch` is now required (Node ≥ 18, modern browsers, Deno, Bun). For runtimes without it, supply a ponyfill via the `alternateFetch` option, or stay on v3.
